storage/src/main/java/io/delta/storage/S3SingleDriverLogStore.java

```diff
@@ -223,8 +236,17 @@ private Iterator<FileStatus> listFromInternal(
             );
         }
 
+        FileStatus[] statuses;
+        if (fs instanceof LocalFileSystem || fs instanceof RawLocalFileSystem // These are needed for tests to pass
+                || !enableFastListFrom
+        ) {
+            statuses = fs.listStatus(parentPath);
+        } else {
+            statuses = S3LogStoreUtil.s3ListFromArray(fs, resolvedPath, parentPath);
```

Comment:
so if this flag is set but they are NOT using S3A file system, then we will throw an UnsupportedOperationException?

This seems wrong to me. If we can easily avoid a failure, then why not do so? Didn't we say to first check if it is castable? Forgive me if my memory serves incorrect.

Reply:
cc @zsxwing

Reply:
We can have two alternative behaviours when the user has enabled the feature flag, but is not using the S3AFileSystem:

1. Throw an exception, and let the user fix the problem. The user will either have to disable the feature flag, or change their S3AFileSystem.
2. Log a warning, and fall back to the old implementation. The user might believe that the feature is enabled, while in fact it is not.

In [this thread](https://github.com/delta-io/delta/pull/1210#discussion_r977894364) we briefly discussed the merits of the two alternatives.

Reply:
Thanks for clarifying.  Option 1 SGTM. Throw the exception and force them to fix the problem.

Sorry for the confusion! Glad you were able to remind me and re-convince me :p